spi/gnap/internal/digest/digest.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package digest
+
+import (
+	"crypto"
+	_ "crypto/sha256"
+	"errors"
+)
+
+const (
+	SHA256 = "sha-256"
+)
+
+// Digest computes the content-digest of the given message.
+type Digest func([]byte) ([]byte, error)
+
+// GetDigest returns the Digest matching the given name.
+func GetDigest(name string) (Digest, error) {
+	switch name {
+	case SHA256:
+		return hashToDigest(crypto.SHA256), nil
+	default:
+		return nil, errors.New("unsupported digest")
+	}
+}
+
+func hashToDigest(hash crypto.Hash) Digest {
+	return func(msg []byte) ([]byte, error) {
+		h := hash.New()
+		_, err := h.Write(msg)
+		if err != nil {
+			return nil, err
+		}
+
+		return h.Sum(nil), nil
+	}
+}
```

Comment:
this code seems familiar

Reply:
Is there something similar in afgo?

I'm not familiar with any, I just wrote this up myself :'D

Reply:
a more generalized (not limited to SHA256) hashing and signing is found [here](https://github.com/hyperledger/aries-framework-go/blob/0ba34929e05b0061b864a4d0c3c08f445ac53ab5/pkg/doc/util/signature/internal/signer/ecdsa.go#L132).. the issue is we have this elsewhere as well. I believe Sidetree also has a copy of this code.. and of course Tink hides all this logic from the user (and by inference, the KMS/Crypto api service as well).